master/internal/resourcemanagers/agent_resource_manager.go

```diff
@@ -500,3 +508,39 @@ func (a *agentResourceManager) createResourcePoolSummary(
 
 	return resp, nil
 }
+
+func (a *agentResourceManager) fetchAvgQueuedTime(pool string) (
+	[]*jobv1.AggregateQueueStats, error,
+) {
+	aggregates := make([]model.ResourceAggregates, 0)
+	err := db.Bun().NewSelect().Model(&aggregates).Where(
+		"aggregation_type = ? AND aggregation_key = ? AND date >= CURRENT_TIMESTAMP - interval '30 days'",
+		"queued", pool,
```

Comment:
also somewhat of a style nitpick. one of the benefits of bun is being able to do chaining, e.g. for `.Where` calls. 
in my opinion one clause per parametrized where plus one or more clause for "fixed" filters is more readable than one clause with multiple `?` and multiple parameters, because the latter leaves you figuring out which parameter goes where. i.e.
```suggestion
		Where("aggregation_type = ?", "queued").
		Where(aggregation_key = ?", pool).
		Where(date >= CURRENT_TIMESTAMP - interval '30 days'").
		Order...
```

Reply:
Thanks! Fixed.